pyramid_hypernova/plugins.py

```diff
@@ -56,15 +56,16 @@ def transform_request_headers(self, headers, request):
             headers = plugin.transform_request_headers(headers, request)
         return headers
 
-    def should_send_request(self, jobs):
+    def should_send_request(self, jobs, request):
         """An every type function. If one returns false then the request is
         canceled.
 
         :type jobs: Dict[str, Job]
         :returns: False if the request should be cancelled
+        :type request: a Pyramid request object
         :rtype: bool
         """
-        return all(plugin.should_send_request(jobs) for plugin in self.plugins)
+        return all(plugin.should_send_request(jobs, request) for plugin in self.plugins)
 
     def will_send_request(self, jobs):
```

Comment:
I think we'll eventually want to pass in the pyramid request to all of these lifecycle methods. Can we do that in this review to avoid multiple breaking releases?

Reply:
Sounds good, I passed in the request objects from the lifecycle methods in the new commit.